src/common/pure/CurrencySelectButton/index.tsx

```diff
@@ -1,14 +1,15 @@
 import { Currency } from '@uniswap/sdk-core'
 
 import { Trans } from '@lingui/macro'
+import { Nullish } from 'types'
 
 import { CurrencyLogo } from 'common/pure/CurrencyLogo'
 import { TokenSymbol } from 'common/pure/TokenSymbol'
 
 import * as styledEl from './styled'
 
 export interface CurrencySelectButtonProps {
-  currency?: Currency
+  currency?: Nullish<Currency>
```

Comment:
do you still need the question-mark in the prop?

Reply:
Yes. `Optional !== undefined`

Reply:
fine with me, different is subtle, the point is if we want to make it optional.

Currency is even in the name of the component, and the 2 uses provide the prop

https://github.com/cowprotocol/cowswap/blob/37d4879d597d7e59df1511eb867a626ef7419d0b/src/common/pure/CurrencyInputPanel/CurrencyInputPanel.tsx#L137

https://github.com/cowprotocol/cowswap/blob/37d4879d597d7e59df1511eb867a626ef7419d0b/src/modules/limitOrders/pure/ReceiptModal/CurrencyField.tsx#L20

Reply:
Aaaah, got your point. Will refactor it in the next PR.
Thank you!